### PR TITLE
Added link test expectations, affects [Dependabot GitHub Twitter]

### PR DESCRIPTION
**Pull request by PyvesB**
See #3242.

I was not sure whether to use the word link as singular or plural. Even though it makes slightly less sense to me, I went for `link` in the end, as this is the form used in [base.js](https://github.com/badges/shields/blob/2806eb8a00bf2b40db5ce3e6910a0b58a1a6fc8e/core/base-service/base.js#L52).

### CODE REVIEW

File: gh-badges/lib/make-badge.spec.js [hunk 1]
```diff
@@ -94,7 +94,11 @@ describe('The badge generator', function() {
 
   describe('JSON', function() {
     it('should always produce the same JSON (unless we have changed something!)', function() {
-      const json = makeBadge({ text: ['cactus', 'grown'], format: 'json' })
+      const json = makeBadge({
+        text: ['cactus', 'grown'],
+        format: 'json',
+        links: ['https://example.com/', 'https://other.example.com/'],
```
Comment (chris48s) [inline gh-badges/lib/make-badge.spec.js]: should this be `link:`, or is this one different on purpose?

File: gh-badges/templates/default-template.json [hunk 2]
```diff
@@ -5,5 +5,6 @@
   "message": {{=JSON.stringify(it.text[1])}},{{?it.logoWidth}}
   "logoWidth": {{=JSON.stringify(it.logoWidth)}},{{?}}{{?it.labelColor}}
   "labelColor": {{=JSON.stringify(it.labelColor)}},{{?}}
-  "color": {{=JSON.stringify(it.color || null)}}
+  "color": {{=JSON.stringify(it.color || null)}}{{?(it.links[0] && it.links[0].length || it.links[1] && it.links[1].length)}},
```
Comment (chris48s) [inline gh-badges/templates/default-template.json]: Should the comma on this line be inbetween the 2 template expressions, like it is on the previous lines?

File: gh-badges/lib/make-badge.spec.js (hunk 1 above)
Comment (PyvesB) [reply gh-badges/lib/make-badge.spec.js]: This one is different on purpose, as the plural form is used in [make-badge.js](https://github.com/badges/shields/blob/75def85eca5f045b147191e8142f3ce146051c4f/gh-badges/lib/make-badge.js#L107).

File: gh-badges/templates/default-template.json (hunk 2 above)
Comment (PyvesB) [reply gh-badges/templates/default-template.json]: `color` may be the final element in the JSON payload if there isn't a `link` element, therefore the comma has to be part of the last template expression. 😉

File: gh-badges/lib/make-badge.spec.js (hunk 1 above)
Comment (paulmelnikow) [reply gh-badges/lib/make-badge.spec.js]: We can fix that as part of #3370 😀

**Pull request merged**